### ISSUE TITLE
cannot load filebrowser through iframe

### ISSUE THREAD

**Issue by shyal** [filebrowser/filebrowser#550]
Hi,

filebrowser refuses to load through an iframe. Would this be a big change?

Thanks,

Shyal

**Comment by eine** [comment]
Hello @shyal. Could you please provide some info about your context? How are you trying to load it? What alternatives have you considered?

**Comment by shyal** [comment]
Hi @1138-4EB. My context is that i am building a lightweight IDE, that'll at most have an Editor (CodeMirror), a Terminal (gotty) and a FileBrowser (filebrowser). I'm using an <iframe> tag for gotty and filebrowser, and the issue is, filebrowser won't load through an iframe.

I tried removing the x-frame-options header:

```diff
--- a/lib/http/http.go
+++ b/lib/http/http.go
@@ -85,7 +85,7 @@ func serve(c *fb.Context, w http.ResponseWriter, r *http.Request) (int, error) {
        }

        // Any other request should show the index.html file.
-       w.Header().Set("x-frame-options", "SAMEORIGIN")
+       // w.Header().Set("x-frame-options", "SAMEORIGIN")
        w.Header().Set("x-content-type-options", "nosniff")
        w.Header().Set("x-xss-protection", "1; mode=block")
```

however it leads to Chrome displaying an error about filebrowser returning the incorrect response.

**Comment by hitmax117** [comment]
I have the same problem in Organizr (iframe as well). Even if I set `proxy_hide_header       x-frame-options;` in nginx I keep getting the broken page when loading in an iframe `Refused to display 'https://URL/' in a frame because it set 'X-Frame-Options' to 'sameorigin'.` So it's a filebrowser specific problem.

**Comment by AciDCooL** [comment]
̶I̶ ̶h̶a̶v̶e̶ ̶t̶h̶e̶ ̶s̶a̶m̶e̶ ̶i̶s̶s̶u̶e̶ ̶t̶h̶r̶o̶u̶g̶h̶ ̶t̶r̶a̶e̶f̶i̶k̶ ̶a̶s̶ ̶p̶r̶o̶x̶y̶ ̶a̶n̶d̶ ̶t̶h̶e̶ ̶o̶f̶f̶i̶c̶i̶a̶l̶ ̶d̶o̶c̶k̶e̶r̶ ̶i̶m̶a̶g̶e̶ ̶(̶d̶o̶c̶k̶e̶r̶ ̶p̶u̶l̶l̶ ̶f̶i̶l̶e̶b̶r̶o̶w̶s̶e̶r̶/̶f̶i̶l̶e̶b̶r̶o̶w̶s̶e̶r̶)̶ ̶i̶n̶ ̶a̶n̶ ̶i̶f̶r̶a̶m̶e̶ ̶w̶i̶t̶h̶i̶n̶ ̶t̶h̶e̶ ̶o̶r̶g̶a̶n̶i̶z̶r̶ ̶t̶a̶b̶s̶,̶ ̶s̶o̶ ̶i̶t̶'̶s̶ ̶p̶r̶o̶b̶a̶b̶l̶y̶ ̶f̶i̶l̶e̶b̶r̶o̶w̶s̶e̶r̶ ̶s̶p̶e̶c̶i̶f̶i̶c̶.̶
̶U̶s̶i̶n̶g̶ ̶t̶h̶e̶s̶e̶ ̶l̶a̶b̶e̶l̶s̶ ̶w̶h̶i̶c̶h̶ ̶w̶o̶r̶k̶ ̶w̶i̶t̶h̶ ̶m̶o̶s̶t̶ ̶c̶o̶n̶t̶a̶i̶n̶e̶r̶s̶ ̶b̶u̶t̶ ̶n̶o̶t̶ ̶w̶i̶t̶h̶ ̶f̶i̶l̶e̶b̶r̶o̶w̶s̶e̶r̶.̶
̶`̶`̶`̶
̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶t̶r̶a̶e̶f̶i̶k̶.̶f̶r̶o̶n̶t̶e̶n̶d̶.̶h̶e̶a̶d̶e̶r̶s̶.̶S̶S̶L̶R̶e̶d̶i̶r̶e̶c̶t̶:̶ ̶"̶t̶r̶u̶e̶"̶
̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶t̶r̶a̶e̶f̶i̶k̶.̶f̶r̶o̶n̶t̶e̶n̶d̶.̶p̶a̶s̶s̶H̶o̶s̶t̶H̶e̶a̶d̶e̶r̶:̶ ̶"̶t̶r̶u̶e̶"̶
̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶t̶r̶a̶e̶f̶i̶k̶.̶f̶r̶o̶n̶t̶e̶n̶d̶.̶h̶e̶a̶d̶e̶r̶s̶.̶f̶r̶a̶m̶e̶D̶e̶n̶y̶:̶ ̶"̶t̶r̶u̶e̶"̶
̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶t̶r̶a̶e̶f̶i̶k̶.̶f̶r̶o̶n̶t̶e̶n̶d̶.̶h̶e̶a̶d̶e̶r̶s̶.̶c̶u̶s̶t̶o̶m̶F̶r̶a̶m̶e̶O̶p̶t̶i̶o̶n̶s̶V̶a̶l̶u̶e̶:̶ ̶"̶a̶l̶l̶o̶w̶-̶f̶r̶o̶m̶ ̶h̶t̶t̶p̶s̶:̶/̶/̶w̶w̶w̶.̶d̶o̶m̶a̶i̶n̶.̶c̶o̶m̶"̶
̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶ ̶t̶r̶a̶e̶f̶i̶k̶.̶f̶r̶o̶n̶t̶e̶n̶d̶.̶h̶e̶a̶d̶e̶r̶s̶.̶c̶o̶n̶t̶e̶n̶t̶S̶e̶c̶u̶r̶i̶t̶y̶P̶o̶l̶i̶c̶y̶:̶ ̶"̶f̶r̶a̶m̶e̶-̶a̶n̶c̶e̶s̶t̶o̶r̶s̶ ̶'̶s̶e̶l̶f̶'̶ ̶d̶o̶m̶a̶i̶n̶.̶c̶o̶m̶ ̶*̶.̶d̶o̶m̶a̶i̶n̶.̶c̶o̶m̶ ̶;̶"̶
̶`̶`̶`̶
̶
̶E̶r̶r̶o̶r̶:̶
̶`̶R̶e̶f̶u̶s̶e̶d̶ ̶t̶o̶ ̶d̶i̶s̶p̶l̶a̶y̶ ̶'̶h̶t̶t̶p̶s̶:̶/̶/̶f̶i̶l̶e̶b̶r̶o̶w̶s̶e̶r̶.̶d̶o̶m̶a̶i̶n̶.̶c̶o̶m̶/̶'̶ ̶i̶n̶ ̶a̶ ̶f̶r̶a̶m̶e̶ ̶b̶e̶c̶a̶u̶s̶e̶ ̶i̶t̶ ̶s̶e̶t̶ ̶'̶X̶-̶F̶r̶a̶m̶e̶-̶O̶p̶t̶i̶o̶n̶s̶'̶ ̶t̶o̶ ̶'̶s̶a̶m̶e̶o̶r̶i̶g̶i̶n̶'̶.̶`̶

Edit:
@̶h̶i̶t̶m̶a̶x̶1̶1̶7̶ ̶T̶r̶y̶ ̶c̶l̶e̶a̶r̶i̶n̶g̶ ̶c̶a̶c̶h̶e̶ ̶i̶n̶ ̶y̶o̶u̶r̶ ̶b̶r̶o̶w̶s̶e̶r̶ ̶o̶r̶ ̶g̶o̶ ̶i̶n̶c̶o̶g̶n̶i̶t̶o̶.̶ ̶T̶h̶a̶n̶k̶s̶ ̶t̶o̶:̶ ̶h̶t̶t̶p̶s̶:̶/̶/̶g̶i̶t̶h̶u̶b̶.̶c̶o̶m̶/̶c̶a̶u̶s̶e̶f̶x̶/̶O̶r̶g̶a̶n̶i̶z̶r̶/̶i̶s̶s̶u̶e̶s̶/̶4̶0̶0̶#̶i̶s̶s̶u̶e̶c̶o̶m̶m̶e̶n̶t̶-̶3̶1̶0̶2̶5̶2̶7̶1̶2̶


Edit2: NVM

Due to the service worker in chrome as far as I can debug. Have two working options.

1. Remove the service worker entirely and iframe works once, since after a reload some js script (sw.js?) adds the worker again the response header x-frame-options goes back to sameorigin.

2. Semi permanent workaround for testing:
Chrome --> F12 --> Application tab --> Service Workers --> check "Bypass for network" --> refresh iframe

Headers
With service worker:
```
content-length: 3365
content-type: text/html; charset=utf-8
date: Sat, 08 Dec 2018 23:54:35 GMT
status: 200
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
x-xss-protection: 1; mode=block
```

Without service worker:
```
content-length: 3365
content-security-policy: frame-ancestors 'self' domain.com *.domain.com ;
content-type: text/html; charset=utf-8
date: Fri, 21 Dec 2018 00:37:11 GMT
status: 200
x-content-type-options: nosniff
x-frame-options: allow-from https://www.domain.com
x-xss-protection: 1; mode=block
```

**Comment by hacdias** [comment]
We have no reason to block the embedding of FIle Browser through iframes. I have removed that HTTP header to allow so. WIll be available on 2.0
